ts/elevenlabs/elevenlabs/elevenlabs.ts

```diff
@@ -0,0 +1,118 @@
+import { ElevenLabsClient } from "elevenlabs";
+import { api } from "encore.dev/api";
+import { IncomingMessage } from "node:http";
+import { secret } from "encore.dev/config";
+import { APICallMeta, currentRequest } from "encore.dev";
+
+const elevenLabsAPIKey = secret("ElevenLabsAPIKey");
+
+const elevenlabs = new ElevenLabsClient({
+  apiKey: elevenLabsAPIKey(),
+});
+
+export const generateVoiceAudio = api.raw(
+  { expose: true, path: "/generate", method: "POST" },
+  async (req, resp) => {
+    const { voice, text, voice_settings } = await getJSONBody<{
+      text: string;
+      voice: string;
+      voice_settings: {
+        stability?: number;
+        similarity?: number;
+        exaggeration?: number;
+      };
+    }>(req);
+
+    resp.setHeader("Content-Type", "audio/mpeg");
+
+    const audioStream = await elevenlabs.generate({
+      stream: true,
+      voice,
+      text,
+      model_id: "eleven_multilingual_v2",
+      voice_settings: {
+        stability: voice_settings.stability || 0.5,
+        similarity_boost: voice_settings.similarity || 0.75,
+        style: voice_settings.exaggeration || 0,
+        use_speaker_boost: true,
+      },
+    });
+
+    audioStream.on("error", console.error);
```

Comment:
Maybe (with `encore.dev/log` imported):

```suggestion
    audioStream.on("error", (err) => log.error("audio stream failed", {err}));
```